file_formats/nwb.md

```diff
@@ -10,15 +10,17 @@ Neurodata Without Borders (NWB) files are a standard file format for neurophysio
 
 ## Basics
 
-We also take this opportunity to remind readers about some NWB basics:
+We also take this opportunity to remind readers about some general NWB basics:
 
-- raw data goes in the `acquisition` group
-- processed data goes in the `processing` group
-- all timestamps are aligned to a single, common clock
+- Raw data goes in the `acquisition` group
+- Processed data goes in the `processing` group
+- The root level `session_start_time` is a timezone-aware datetime defining the start of acquisition.
+- All timestamps must relative to the same point in time and require no additional alignment to compare. This is defined in the root level `timestamps_reference_time`, which defaults to `session_start_time`.
+- It is not required that timestamps for different data streams be identical (i.e. resample data such that all timeseries to have exactly the same timestamps).
 
-## Timestamps
+## HARP
 
-- If there is a harp board available, all timestamps will be aligned to the harp clock.
+- At AIND, if there is a HARP board available, timestamps must utilize HARP TTLs to transform timestamps to a common timebase. It is insufficient to store a lookup table.
 
```

Comment:
Do we need to add other hardware types?

Reply:
Is it actually relevant what the "reference clock" is at the level of NWB (prior to data package for sure)? We are aligning everything to harp, but one could also align everything to ephys, from the point of view of a data consumer, does it matter? The assumption is that everything is in a stable temporal format anyway.

Reply:
NWB only requires that timestamps are directly comparable (point made above).

I additionally strongly believe that deferring temporal alignment to after processing makes it extremely difficult to automate NWB generation and therefore sharing. I don't have a place to write this down right now, so I'm just strongly implying it here.

Reply:
I am reading this as being agnostic to hardware and that the timestamps from the reference clock used in the original system design should be added to the time stamps field. Meaning if the common clock for the system was produced by harp, sync or bpod, it doesn't matter. We just want the common clock times in the time stamp field. I was confused why we were calling out Harp specifically. What if the hardware changes in the future?

Reply:
This is why I prefix with "At AIND" - this is articulating that we are adopting harp as a preferred standard.